core-bundle/src/Resources/contao/library/Contao/StringUtil.php

```diff
@@ -871,9 +871,11 @@ static function ($match)
 			. ');?'               // Optional semicolon
 		. ')i';
 
+		$arrAllowedProtocols = System::getContainer()->getParameter('contao.allowed_protocols');
+
 		// URL-encode colon to prevent disallowed protocols
 		if (
-			!preg_match('@^(?:https?|ftp|mailto|tel|data):@i', self::decodeEntities($strString))
+			!preg_match('@^(?:' . implode('|', array_map('preg_quote', $arrAllowedProtocols)) . '):@i', self::decodeEntities($strString))
```

Comment:
```suggestion
			!preg_match('(^(?:' . implode('|', array_map('preg_quote', $arrAllowedProtocols)) . '):)i', self::decodeEntities($strString))
```

Default delimiters need to be used together with `preg_quote()`